src/helpers.ts

```diff
@@ -371,18 +371,61 @@ export function isActionTrigger(ev: KeyboardEvent): boolean {
     }
 }
 
-/*
+/**
  * Create a shallow copy of an object.
  */
-export function copyShallow<T extends object>(obj: T): T {
-    return Object.assign({}, obj);
+export function copyShallow(object: object): object {
+    return Object.assign({}, object);
 }
 
 /**
- * Create a deep copy of an object by serializing and deserializing it.
+ * Create a deep copy.
+ *
+ * This handles the following types:
+ *
+ * - copies `undefined` and `null`,
+ * - copies `Boolean`, `Number` and `String`,
+ * - copies `object` recursively,
+ * - copies `Array` recursively,
+ * - copies `ArrayBuffer`,
+ * - copies `Uint8Array`,
+ *
+ * Everything else will be **referenced**.
```

Comment:
Isn't that totally against the intent of a deep copy?

Reply:
Considering that it was not copied before either but just represented as an empty object, this already is an improvement. I'm happy to bikeshed on the name. Suggestions?

Reply:
You have to see it in the context we're using it. That is currently solely for logging binary data that could be modified later. This is not meant to be a general purpose deep copy function that covers all edge cases. The catch-all currently handles copying of all non-special types. Throwing an exception instead (which is an option) would require checking more types.

Reply:
That's fine, but then you need to change the name. I don't have a good idea though, `copySemideep`? :stuck_out_tongue: 

The fallback could also be serialize-deserialize, as we did for the entire object previously.

Reply:
> The fallback could also be serialize-deserialize, as we did for the entire object previously.

That adds zero value since it just returns an empty object.

Reply:
Only for non-serializable types (like ArrayBuffer), of which we explicitly handle a few. But I'd prefer the renaming too.

`copyDeeper` could also be a potential name.

Reply:
> Only for non-serializable types

There aren't any further types. JSON is, by definition, limited to JSON.

How about `copyDeepish`?

Reply:
`copyMost`?

Reply:
`copyDeepExceptWhereItReferences`.